party_time/party_time.lua

```diff
@@ -0,0 +1,341 @@
+local treasure = require('treasure')
+
+local defaults = {
+    ui = {
+        x = 145,
+        y = 440,
+        enabled = true,
+    },
+    auto = {
+        accept = true,   -- also for auto sending invites upon party requests
+        decline = false, -- also ignores party requests from players on blacklists
+    },
+    blacklist = sets({}),
+    whitelist = sets({}),
+}
+settings.settings_change:register(function(options)
+    options.blacklist = sets(options.blacklist)
+    options.whitelist = sets(options.whitelist)
+end)
+options = settings.load(defaults)
+
+local invite_dialog = {}
+local invite_dialog_state = {
+    title = 'Party Invite',
+    style = 'normal',
+    x = options.ui.x,
+    y = options.ui.y,
+    width = 179,
+    height = 96,
+    resizable = false,
+    moveable = true,
+    closable = true,
+}
+local unhandled_requests = {}
+local unhandled_invite = false
+
+-- Command Arg Types
+-- Create and Register types
+local arg_lookups = {
+    addremove = {
+        ['add'] = 'union',
+        ['a'] = 'union',
+        ['+'] = 'union',
+        ['remove'] = 'difference',
+        ['rm'] = 'difference',
+        ['r'] = 'difference',
+        ['-'] = 'difference',
+    },
+    bool = {
+        ['1'] = true,
+        ['true'] = true,
+        ['t'] = true,
+        ['yes'] = true,
+        ['y'] = true,
+        ['on'] = true,
+        ['0'] = false,
+        ['false'] = false,
+        ['f'] = false,
+        ['no'] = false,
+        ['n'] = false,
+        ['off'] = false,
+    },
+}
+
+command.arg.register('name', '<name:string(%a+)>')
+
+command.arg.register_type('lookup_boolean', {
+    check = function(str)
+        local bool = arg_lookups.bool[str]
+        if bool ~= nil then
+            return bool
+        end
+
+        error('Expected one of \'' .. table.concat(args_lookups.bool, ',') .. '\', received \'' .. str .. '\'.')
+    end
+})
+
+command.arg.register_type('lookup_add_remove', {
+    check = function(str)
+        local op = arg_lookups.addremove[str]
+        if op ~= nil then
+            return op
+        end
+
+        error('Expected one of \'' .. table.concat(args_lookups.addremove, ',') .. '\', received \'' .. str .. '\'.')
+    end
+})
+
+-- Addon Command Handlers
+-- Seven main commands additoinal sub_commands within
+local pt = command.new('pt')
+
+local invite = function(...)
+    for _, name in pairs({...}) do
+        command.input('/pcmd add ' .. name)
+        name = name:gsub('^%l', string.upper)
+        if unhandled_requests[name] then
+            unhandled_requests[name] = nil
+        end
+    end
+end
+
+pt:register('i', invite, '{name}*')
+pt:register('invite', invite, '{name}*')
+
+
+local request = function(name)
+    command.input('/prcmd add ' .. name)
+end
+
+pt:register('r', request, '{name}')
+pt:register('request', request, '{name}')
+
+
+local blacklist = function(sub_cmd, ...)
+    local names = sets({...})
+    options.blacklist[sub_cmd](options.blacklist, names)
```

Comment:
While it's not really an issue, this could be `set[sub_cmd](options.blacklist, names)`, so you don't have to type `options.blacklist` out twice. Syntactically what you have is just a "shortcut" for `set[...]`, which in this case is actually longer.